### PR TITLE
[Documentation][Locale] Component docs fix

### LINKED ISSUE #8811: [Documentation] Locale component missing docs
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Docs request? | yes
| BC Break report? | no
| RFC?             | no
| Sylius version   | 1.0

* title should be `Locale` instead of `Locales`
* at least the `CompositeLocaleContext` should be described here in place of the empty `LocaleContext` paragraph - we should add that it has the `getLocaleCode` method, and that it is a composite of different contexts which are prioritized in it (the one with highest priority is used if it exists)

* we can also mention that the following contexts reside in the Locale component: `ImmutableLocaleContext` and `ProviderBasedLocaleContext`

### PR DESCRIPTION
**Pull request by CoderMaggie**
| Q               | A
| --------------- | -----
| Branch?         | 1.0 
| Doc fix?        | yes
| Related tickets | #8813 , fixes #8811
| License         | MIT

Changes thanks to @nightrainlily who is our **first time contributor** 🔥 ! I just made a rebase here :D

### CODE REVIEW

File: docs/components_and_bundles/components/Locale/basic_usage.rst [hunk 1]
```diff
@@ -4,7 +4,16 @@ Basic Usage
...
+In the Locale component there are three LocaleContexts defined:
+* ``CompositeLocaleContext``
+* ``ImmutableLocaleContext``
+* ``ProviderBasedLocaleContext``
+
+CompositeLocaleContext
+~~~~~~~~~~~~~~~~~~~~~~
+
+	It is a composite of different contexts available in your application, which are prioritized while being injected here (the one with highest priority is used).
+	It has the ``getLocaleCode``, ``_construct()``, and ``addContext(LocaleContextInterface $localeContext, int $priority)`` methods available.
```
Comment (pamil) [inline docs/components_and_bundles/components/Locale/basic_usage.rst]: This line is quite fragile and can be replaced by looking at the source code instead.

Comment (pamil) [inline docs/components_and_bundles/components/Locale/basic_usage.rst]: Anyway, `_construct()` -> `__construct()` and `getLocaleCode` -> `getLocaleCode()` if we decide to keep it

Comment (CoderMaggie) [reply docs/components_and_bundles/components/Locale/basic_usage.rst]: Fixed :)

**Comment by pamil** [comment]
Thank you @nightrainlily and @CoderMaggie! 👍

**Pull request merged**